src/util/permissions.ts

```diff
@@ -1,249 +1,340 @@
-  if (permissions.every((perm) => allowedPermissions.has(perm))) return true;
+/** Throws an error if the bot does not have all permissions */
+export function requireBotGuildPermissions(
+  guild: string | Guild,
+  permissions: Permission[],
+) {
+  // Since Bot is a normal member we can use the throwOnMissingGuildPermission() function
+  return requireGuildPermissions(guild, botID, permissions);
+}
+
+/** Throws an error if this member has not all of the given permissions */
+export async function requireChannelPermissions(
+  channel: string | Channel,
+  member: string | Member,
+  permissions: Permission[],
+) {
+  const missing = await getMissingChannelPermissions(
+    channel,
+    member,
+    permissions,
+  );
+  if (missing.length) {
+    // If the member is missing a permission throw an Error
+    throw new Error(`Missing Permissions: ${missing.join(" & ")}`);
+  }
+}
 
-  // Some permission was not explicitly allowed so we default to checking role perms directly
-  return memberIDHasPermission(memberID, guild.id, permissions);
+/** Throws an error if the bot has not all of the given channel permissions */
+export function requireBotChannelPermissions(
+  channel: string | Channel,
+  permissions: Permission[],
+) {
+  // Since Bot is a normal member we can use the throwOnMissingChannelPermission() function
+  return requireChannelPermissions(channel, botID, permissions);
 }
 
 /** This function converts a bitwise string to permission strings */
 export function calculatePermissions(permissionBits: bigint) {
-  return Object.keys(Permissions).filter((perm) => {
-    if (Number(perm)) return false;
-    return permissionBits & BigInt(Permissions[perm as Permission]);
+  return Object.keys(Permissions).filter((permission) => {
+    // Since Object.keys() not only returns the permission names but also the bit values we need to return false if it is a Number
+    if (Number(permission)) return false;
+    // Check if permissionBits has this permission
+    return permissionBits & BigInt(Permissions[permission as Permission]);
   }) as Permission[];
 }
 
 /** This function converts an array of permissions into the bitwise string. */
 export function calculateBits(permissions: Permission[]) {
-  return permissions.reduce(
-    (bits, perm) => bits |= BigInt(Permissions[perm]),
-    BigInt(0),
-  ).toString();
+  // return permissions
+  //   .reduce(
+  //     // Get the bit value for this permission and assign it to bits
+  //     (bits, perm) => {
+  //       bits |= BigInt(Permissions[perm]);
+  //         return bits
+  //       }),
+  //     0n,
+  //   )
+  //   .toString();
+
+  return permissions
+    .reduce((bits, perm) => {
+      bits |= BigInt(Permissions[perm]);
+      return bits;
+    }, 0n)
+    .toString();
 }
 
-export async function highestRole(guildID: string, memberID: string) {
-  const guild = await cacheHandlers.get("guilds", guildID);
-  if (!guild) return;
+// TODO: move memberID to first position
```

Comment:
member should always be after guild. It should follow this order: guild > channel/role/member > message

Reply:
Yes, I forgot to delete this comment.